src/core/symbology/qgsmarkersymbollayer.cpp

```diff
@@ -1178,7 +1178,11 @@ bool QgsSimpleMarkerSymbolLayer::prepareCache( QgsSymbolRenderContext &context )
     return false;
   }
 
-  mCache = QImage( QSize( imageSize, imageSize ), QImage::Format_ARGB32_Premultiplied );
+  mCache = QImage( QSize( imageSize * context.renderContext().devicePixelRatio(),
```

Comment:
Shouldn't this new imageSize be tested for MAXIMUM_CACHE_WIDTH ? I.e. move the check after the image size is multiplied by devicePixelRatio ?

Reply:
Correct, fixed